src/useObserver.ts

```diff
@@ -67,7 +67,9 @@ export function useObserver<T>(
                 // We haven't yet reached useEffect(), so we'll need to trigger a re-render
                 // when (and if) useEffect() arrives.  The easiest way to do that is just to
                 // drop our current reaction and allow useEffect() to recreate it.
-                newReaction.dispose()
+                reactionTrackingRef.current!.reaction.dispose()
+                // @ts-ignore
+                reactionTrackingRef.current!.reaction = null
```

Comment:
This just doesn't feel like it's necessary. I mean it won't probably harm anything, but I would rather keep code that makes sense only. It's obvious, that calling `dispose` won't remove reaction from the memory immediately. Even removing its reference doesn't give such assurance, it's up to GC.

Reply:
@hzzcc Can you respond to this? Not sure why you marked it as resolved without an answer.

Reply:
@FredyC Yes, this is up to GC, but  can take heap snapshot by youself, click several times until memory stabilizes.

Reply:
I don't follow. How does that explain why do you need to set a reaction to null here?

Reply:
Dereference 'reaction'. But, I guess input change event cause these. when I change to 'div', the problem disappear.

Reply:
Ok, so please remove this from PR if it's not necessary, it only brings confusion.